sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala

```diff
@@ -276,7 +297,7 @@ case class EnsureRequirements(conf: SQLConf) extends Rule[SparkPlan] {
    * introduced). This rule will change the ordering of the join keys to match with the
    * partitioning of the join nodes' children.
    */
-  private def reorderJoinPredicates(plan: SparkPlan): SparkPlan = {
```

Comment:
In the historical reason (https://github.com/apache/spark/pull/19257#issuecomment-345278009), this method was added as a workaround. So, I feel it is compliated to extend this method for this case... basically, IMO we need a general logic here to conver this case and more. cc: @cloud-fan